File: splearn-java/src/main/java/io/wisoft/splearn/application/member/provided/MemberRegister.java
package io.wisoft.splearn.application.member.provided;

import io.wisoft.splearn.domain.member.Member;
import io.wisoft.splearn.domain.member.MemberInfoUpdateRequest;
import io.wisoft.splearn.domain.member.MemberRegisterRequest;
import jakarta.validation.Valid;

/**
 * 회원의 등록과 관련된 기능을 제공한다
 */
public interface MemberRegister {
    Member register(@Valid MemberRegisterRequest registerRequest);

    Member activate(Long memberId);

    Member deactivate(Long memberId);

    Member updateInfo(Long memberId, @Valid MemberInfoUpdateRequest updateRequest);
}

File: splearn-typescript/src/main/application/provided/member-register.ts
/**
 * 회원의 등록과 관련된 기능을 제공한다
 */
import { Member } from "@src/main/domain/member";

import type { MemberRegisterRequest } from "@src/main/domain/member-register-request";

export interface MemberRegister {
  register(registerRequest: MemberRegisterRequest): Member;
}